theano/gof/wrapper.py

```diff
@@ -0,0 +1,446 @@
+            except MethodNotDefined:
+                pass
+        return c_compile_args_list
+
+    def c_no_compile_args(self):
+        c_no_compile_args_list = []
+        for _type in self.types:
+            try:
+                try:
+                    c_no_compile_args_list.extend(_type.c_no_compile_args())
+                except TypeError:
+                    c_no_compile_args_list.extend(_type.c_no_compile_args(compiler))
+            except MethodNotDefined:
+                pass
+        return c_no_compile_args_list
+
+    def c_headers(self):
+        c_headers_list = []
+        for _type in self.types:
+            try:
+                try:
+                    c_headers_list.extend(_type.c_headers())
+                except TypeError:
+                    c_headers_list.extend(_type.c_headers(compiler))
+            except MethodNotDefined:
+                pass
+        return c_headers_list
+
+    def c_libraries(self):
+        c_libraries_list = []
+        for _type in self.types:
+            try:
+                try:
+                    c_libraries_list.extend(_type.c_libraries())
+                except TypeError:
+                    c_libraries_list.extend(_type.c_libraries(compiler))
+            except MethodNotDefined:
+                pass
+        return c_libraries_list
+
+    def c_header_dirs(self):
+        c_header_dirs_list = []
+        for _type in self.types:
+            try:
+                c_header_dirs_list.extend(_type.c_header_dirs())
+            except MethodNotDefined:
+                pass
+        return c_header_dirs_list
+
+    def c_lib_dirs(self):
+        c_lib_dirs_list = []
+        for _type in self.types:
+            try:
+                c_lib_dirs_list.extend(_type.c_lib_dirs())
+            except MethodNotDefined:
+                pass
+        return c_lib_dirs_list
+
+    def c_init_code(self):
+        c_init_code_list = []
+        for _type in self.types:
+            try:
+                c_init_code_list.extend(_type.c_init_code())
+            except MethodNotDefined:
+                pass
+        return c_init_code_list
+
+    def c_support_code(self):
+        sub = {'fail': '{this->setErrorOccurred(); this->cleanup(); return;}'}
+        struct_name = self.name
+        struct_name_defined = struct_name.upper()
+        struct_fields = ''
+        struct_init = ''
+        struct_cleanup = ''
+        struct_extraction_methods = ''
+        c_declare_list = []
+        c_init_list = []
+        c_cleanup_list = []
+        c_extract_list = []
+        for attribute_name, type_instance in zip(self.fields, self.types):
+            type_name = type_instance.__class__.__name__
+
+            try:
+                c_declare_list.append(type_instance.c_declare(attribute_name, sub))
+            except MethodNotDefined:
+                raise RuntimeError('Wrapper: class "%s" should implement method Type.c_declare().' % type_name)
```

Comment:
I don't this we need customized error messages at this point.  Also catching and re-raising exceptions is super annoying when trying to debug with pdb since we don't get the original error location.